HLTrigger/Configuration/python/customizeHLTforAlpaka.py

```diff
@@ -1,7 +1,554 @@
 import FWCore.ParameterSet.Config as cms
 
+
+## Pixel HLT in Alpaka
+def customizeHLTforDQMGPUvsCPUPixel(process):
+    '''Ad-hoc changes to test HLT config containing only DQM_PixelReconstruction_v and DQMGPUvsCPU stream
+       only up to the Pixel Local Reconstruction
+    '''
+    dqmPixelRecoPathName = None
+    for pathName in process.paths_():
+        if pathName.startswith('DQM_PixelReconstruction_v'):
```

Comment:
I am wondering if this approach is a bit brittle. Perhaps one could check if there are the generic product consumers within instead?

Reply:
I agree, I'm not a fan of this too. But I'm not understanding the reference to "generic product consumers". I was thinking one could check if in the path includes the DQM modules we want to replace?

Reply:
in cuda we used to have this:

```
hltPixelConsumerCPU = cms.EDAnalyzer( "GenericConsumer",
    eventProducts = cms.untracked.vstring( 'hltSiPixelDigisLegacy' ),
    lumiProducts = cms.untracked.vstring(  ),
    runProducts = cms.untracked.vstring(  ),
    processProducts = cms.untracked.vstring(  )
)
hltPixelConsumerGPU = cms.EDAnalyzer( "GenericConsumer",
    eventProducts = cms.untracked.vstring( 'hltSiPixelDigis@cuda' ),
    lumiProducts = cms.untracked.vstring(  ),
    runProducts = cms.untracked.vstring(  ),
    processProducts = cms.untracked.vstring(  )
)
```

they are not used anywhere else in the menu.

Reply:
A possible side note: the use of `GenericConsumer` was needed with the old framework based on `SwitchProducer`s because of a limitation of the `ConditionalTask`.

If (once ?) the alpaka-based menu does not use the `SwitchProducer`s, it should be possible to get rid of of the `GenericConsumer`s (and maybe also of the `ConditionalTask`s).